chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/RoomStore.kt

```diff
@@ -30,4 +30,22 @@ internal class RoomStore(
     operator fun minusAssign(roomId: String) {
         roomsMap -= roomId
     }
+}
+
+fun List<Room>.containsRoomId(id: String): Boolean {
+    val iterator = this.iterator()
+    while (iterator.hasNext()) {
+        if (iterator.next().id == id) {
+            return true
+        }
+    }
+
+    return false
+}
+
+operator fun List<Room>.minus(elements: List<Room>): List<Room> {
+    val mutableRooms: MutableList<Room> = this.toMutableList()
+    mutableRooms.removeAll { room -> elements.containsRoomId(room.id) }
+
+    return mutableRooms.toList()
 }
```

Comment:
A more idiomatic way to acheive this would be to override the default `equals(Any?)` on `Room` and make it compare only on `id`. If the default implementation of `minus` on lists compares elements structurally (using `==`, which delegates to `equals`, and asks "are these objects equivalent?") then this will just work.

If it compares elements based on reference (`===`, which asks "are do these references point to the actual same object?") then that's slightly different. https://kotlinlang.org/docs/reference/equality.html

Either way, I think what we want to do is re-define equality for `Room`s rather than redefine removing from `List`.

Reply:
Ah, that does make things way simpler.